woodwork/logical_types.py

```diff
@@ -830,6 +835,23 @@ def _regex_validate(regex_key, series, return_invalid_values):
             raise TypeValidationError(info)
 
 
+def _replace_nans(series: pd.Series) -> pd.Series:
+    """
+    Replaces empty string values, string representations of NaN values ("nan", "<NA>"), and NaN equivalents
+    with np.nan or pd.NA depending on column dtype.
+    """
+    original_dtype = series.dtype
+    if str(original_dtype) == "string":
+        series = series.replace(ww.config.get_option("nan_values"), pd.NA)
+        return series
+    if not _is_spark_series(series):
+        series = series.replace(ww.config.get_option("nan_values"), np.nan)
+    if str(original_dtype) == "boolean":
+        series = series.astype(original_dtype)
+
+    return series
```

Comment:
I think there's already a function that does this. Is [_replace_nan_strings](https://github.com/alteryx/woodwork/blob/5e9b742ec8cc060807323d4c0b5f6daacf6d9659/woodwork/utils.py#L606) useful in this case?

Reply:
That function was removed in favor of `_replace_nans`

Reply:
It relied more on regex and string replacement as opposed to overall nan representation replacement